api/src/game_handlers.rs

```diff
@@ -0,0 +1,655 @@
+use crate::client_structs::{
+    ClientCard, CountedCardResponse, CutCardResponse, CutCards, ParsedHand, RandomHandResponse,
+    ScoreResponse,
+};
+use actix_web::{web::Path, HttpResponse, Responder};
+use cribbage_library::cards::Card;
+use cribbage_library::counting::score_counting_cards_played;
+use cribbage_library::cribbage_errors::{CribbageError, CribbageErrorKind};
+use cribbage_library::scoring::{score_hand as scorehand, Score};
+use cribbage_library::select_cards::{get_next_counted_card, select_crib_cards};
+use rand::prelude::{Rng, SliceRandom};
+
+const HOST_NAME: &'static str = "localhost:8088/api"; // important:  no ending '/'
+
+//
+//  cut the cards to see who goes first
+//
+//  sample URLs:
+//              http://localhost:8080/api/cutcards
+//
+//  returns: the two cut cards and the repeat URL.  the client is written to assume a shared notion of the deck
+//           so we just return 2 numbers bewtween 0 and 51
+//
+pub async fn cut_cards() -> impl Responder {
+    let mut rng = rand::thread_rng();
+    let first = rng.gen_range(0..51) as usize;
+    let mut second: usize;
+    loop {
+        second = rng.gen_range(0..51) as usize;
+        if first != second {
+            break;
+        }
+    }
+    let cc = CutCards::new(first, second);
+    let response = CutCardResponse {
+        CutCards: cc,
+        RepeatUrl: format!("{}/cutcards/{},{}", HOST_NAME, first, second),
+    };
+
+    HttpResponse::Ok().body(serde_json::to_string(&response).unwrap())
+}
+//
+//  cut the cards to see who goes first - pass in the random numbers that you get the same result
+//  the last time the api was called. useful for testing.
+//
+//  sample URLs:
+//              http://localhost:8080/api/cutcards/1,8
+//
+//  returns: the two cut cards
+//
+pub async fn cut_cards_repeat(cards: Path<String>) -> impl Responder {
+    let cards = cards.into_inner();
+    let tokens: Vec<&str> = cards.split(",").collect();
+    if tokens.len() != 2 {
+        return HttpResponse::BadRequest()
+            .body("there should be two cards seperated by a ',' such as '1,2'");
+    }
+    let first = match tokens[0].parse::<usize>() {
```

Comment:
Could you use a tuple to do this with one match? I don't know the exact tuple syntax for this, but it might look something like:

`let (first, second) = match (tokens[0].parse::<usize>(), tokens[1].parse::<usize>()) ...`

Or you could do both parse calls and assignments and then match on the tuple (Ok(first), Ok(second)...

Something like that, should make it more compact and DRY from an error handling perspective.